spec/virtual_includes_spec.rb

```diff
@@ -101,28 +101,35 @@
 
-      # expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
-      # expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
       expect(Author.includes(:books => {:author_name =>{}}).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
     end
 
     it "uses included fields" do
       skip("AR 5.1 not including properly") if ActiveRecord.version.to_s >= "5.1"
-      # expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
-      # expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
       expect(Author.includes(:books => {:author_name => {}}).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
     end
 
     it "uses preloaded fields" do
       skip("AR 5.1 not including properly") if ActiveRecord.version.to_s >= "5.1"
-      # expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
-      # expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => :author_name).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
+      expect(Author.includes(:books => [:author_name]).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
       expect(Author.includes(:books => {:author_name => {}}).references(:books => {:author_name => {}})).to preload_values(:first_book_author_name, author_name)
-      # inc = Author.virtual_includes(:first_book_author_name)
-      # expect(Author.includes(inc).references(inc)).to preload_values(:first_book_author_name, author_name)
+      inc = Author.virtual_includes(:first_book_author_name)
+      expect(Author.includes(inc).references(inc)).to preload_values(:first_book_author_name, author_name)
+    end
+
+    it "detects errors" do
```

Comment:
Can you add one example to this group where it's correct? and change this to "raises if includes don't have a proper uses clause?

Reply:
I think a good example in contrast to the failure ones will better demonstrate what we're doing here.

Reply:
something like this?

```ruby
      expect { Author.includes(:books).references(:books).load}.not_to raise_error
```

Reply:
Right